common/src/java/org/apache/hive/common/util/HiveStringUtils.java

```diff
@@ -1174,4 +1175,25 @@ private static boolean isComment(String line) {
     return lineTrimmed.startsWith("#") || lineTrimmed.startsWith("--");
   }
 
+  /**
+   * Returns integer value of a string. If the string value exceeds max int, returns Integer.MAX_VALUE
+   * else if the string value is less than min int, returns Integer.MIN_VALUE
+   *
+   * @param value value of the input string
+   * @return integer
+   */
+  public static int convertStringToBoundedInt(String value) {
+    try {
+      BigInteger bigIntValue = new BigInteger(value);
+      if (bigIntValue.compareTo(BigInteger.valueOf(Integer.MAX_VALUE)) > 0) {
+        return Integer.MAX_VALUE;
```

Comment:
@vamshikolanu 
I agree with @jfsii. Converting a large number to Integer.MAX_VALUE is misleading the user. 
Consider the following query - 
`INSERT INTO TABLE destinationTable SELECT * FROM sourceTable LIMIT <some_large_number>;`
The insert will write records based on the output of the SELECT operator. In this case, since we have converted it to Integer.MAX_VALUE, the number of records written will be equal to Integer.MAX_VALUE which might not be what the user wants.

Perhaps adding a meaningful exception is better. In the long term, adding support for large integers for LIMIT clauses is even more better.